src/app/components/shared/menu/link/link.component.ts

```
@@ -0,0 +1,111 @@
+          <ng-container *ngTemplateOutlet="linkDetails"></ng-container>
+        </a>
+      </ng-template>
+    </span>
+
+    <!-- Link Details -->
+    <ng-template #linkDetails>
+      <div class="icon"><fa-icon [icon]="link.icon"></fa-icon></div>
+      <span id="label">{{ link.label }}</span>
+      <span class="d-none" [id]="id">{{ tooltip }}</span>
+    </ng-template>
+  `,
+  styleUrls: ["./link.component.scss"],
+  // This will be recreated every time the page loads
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MenuLinkComponent implements OnChanges {
+  @Input() public id: string;
+  @Input() public link: MenuRoute | MenuLink;
+  @Input() public placement: Placement;
+  @Input() public tooltip: string;
+  public active: boolean;
+  public href: string;
+
+  public constructor(
+    @Inject(API_ROOT) private apiRoot: string,
+    private activatedRoute: ActivatedRoute,
+    private location: Location
+  ) {}
+
+  public ngOnChanges() {
+    const params = this.activatedRoute.snapshot.params;
+
+    if (this.isInternalLink) {
+      this.handleInternalLink(params);
+    } else {
+      this.handleExternalLink(params);
+    }
+  }
+
+  public get isInternalLink() {
+    return isInternalRoute(this.link);
+  }
+
+  public get internalLink(): MenuRoute {
+    return this.link as MenuRoute;
+  }
+
+  public get externalLink(): MenuLink {
+    return this.link as MenuLink;
+  }
+
+  private handleInternalLink(params: Params) {
+    const currentRoute = this.internalLink.route.toRouterLink(params);
+
+    // Determine if link is active
+    this.active =
+      this.link.active ||
+      this.link.highlight ||
+      currentRoute === this.location.path().split("?")[0];
+  }
+
+  private handleExternalLink(params: Params) {
+    const uri = this.externalLink.uri(params);
+
+    // Prepend apiRoot to relative links
+    this.href = uri.startsWith("/") ? this.apiRoot + uri : uri;
```

hmmm

this is only correct at the moment? apiRoot should be the route of the API, the external link could be pointing anywhere?

while the old-client is currently hosted on the api server I didn't want to be tied to that. Maybe leave a TODO or a note, this seems like a point of silent failure